ArduPlane/fence.cpp

```diff
@@ -80,17 +81,30 @@ void Plane::fence_check()
         switch (fence_act) {
         case AC_FENCE_ACTION_REPORT_ONLY:
             break;
+
         case AC_FENCE_ACTION_GUIDED:
         case AC_FENCE_ACTION_GUIDED_THROTTLE_PASS:
         case AC_FENCE_ACTION_RTL_AND_LAND:
-            if (fence_act == AC_FENCE_ACTION_RTL_AND_LAND) {
+#if MODE_AUTOLAND_ENABLED
+        case AC_FENCE_ACTION_AUTOLAND_OR_RTL:
+#endif
+            if (fence_act == AC_FENCE_ACTION_RTL_AND_LAND 
+#if MODE_AUTOLAND_ENABLED
+|| fence_act == AC_FENCE_ACTION_AUTOLAND_OR_RTL
+#endif
+                                               ) {
                 if (control_mode == &mode_auto &&
                     mission.get_in_landing_sequence_flag() &&
                     (g.rtl_autoland == RtlAutoland::RTL_THEN_DO_LAND_START ||
                      g.rtl_autoland == RtlAutoland::RTL_IMMEDIATE_DO_LAND_START)) {
                     // already landing
                     return;
```

Comment:
I think this early return needs a check for being in either RTL or Autoland.

Reply:
I have added a check to not change modes on a breach when AUTOLANDING in is LAND stage